testing/trino-tests/src/test/java/io/trino/tests/TestJoinQueries.java

```diff
@@ -42,4 +45,41 @@ public void verifyDynamicFilteringEnabled()
                 "SHOW SESSION LIKE 'enable_dynamic_filtering'",
                 "VALUES ('enable_dynamic_filtering', 'true', 'true', 'boolean', 'Enable dynamic filtering')");
     }
+
+    /**
+     * This test verifies if a broadcast deadlock is getting properly resolved.
+     * <p>
+     * A deadlock can happen when the build side of a join overflows the total capacity of the broadcast output buffer.
+     * When the broadcast buffer is overflow some data must be discarded. The data from the broadcast output buffer can
+     * only be discarded after it is consumed by all consumers. The scheduler is expected to send the "noMoreOutputBuffers"
+     * signal when the probe side scheduling is done. However if the probe side scheduling is blocked on split placement
+     * the scheduling might never finish. To handle this case a special handling was introduced. When the scheduler detects
+     * that the stage is blocked on the split placement and the output buffers of the source tasks of the stage are full the
+     * scheduler schedules as many tasks as there are nodes in the cluster (without waiting for the split placement to finish)
+     * and sends a signal to the source tasks that no more tasks (thus output buffers) will be created.
+     */
+    @Test(timeOut = 600_000)
+    public void testBroadcastJoinDeadlockResolution()
```

Comment:
There are multiple flavours of `AbstractTestJoinQueries`. Would that make sense to make sense to have more that one incarnation of this test (with/without DF ...)

Reply:
The intention was to add a regression test for this specific issue. Also the test is `TPCH` specific, as it relies on the TPCH implementation details to ensure enough splits are produce to trigger a "deadlock".